src/game/tile.h

```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+enum class tile{
```

Comment:
tile is little descriptive name. BoardTile or smth like that would be better. I don't think we will perform all the operations on square board - model is likely to be changed in the future so mixing this with Board is good idea imo.